src/pipeline.ts

```diff
@@ -256,6 +268,76 @@ export class GitHubWorkflow extends PipelineBase {
     return stageDeployment;
   }
 
+  /**
+   * Add a Wave to the pipeline, for deploying multiple Stages in parallel
+   *
+   * Use the return object of this method to deploy multiple stages in parallel.
+   *
+   * Example:
+   *
+   * ```ts
+   * declare const pipeline: GitHubWorkflow; // assign pipeline a value
+   *
+   * const wave = pipeline.addWave('MyWave');
+   * wave.addStage(new MyApplicationStage(this, 'Stage1'));
+   * wave.addStage(new MyApplicationStage(this, 'Stage2'));
+   * ```
+   */
+  public addWave(id: string, options?: WaveOptions): Wave {
+    return this.addGitHubWave(id, options);
+  }
```

Comment:
Hmm we probably just want to direct users to use `addGitHubWave` instead of `addWave` entirely, right? It feels confusing to have both `addWave` and `addGitHubWave` as part of the API.

I think we should consider either throwing an error when you use `addWave` and explaining that you should be using `addGitHubWave`, or at the very least, providing a warning (via `Annotations` construct) that you should be using `addGitHubWave`.

Reply:
Unfortunately I spent a while trying to do this one to determine that it cannot be done, at least not without wrecking other things.

`PipelineBase.addStage` calls `addWave` internally. We'd have to rewrite `addStage` as well. I don't think it's worth it.

https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/pipelines/lib/main/pipeline-base.ts#L92-L97

I feel like the real solution would be upstream (adjust `PipelineBase`, etc, but I can't determine the best route at the moment.)